src/AlertBox/AlertBox.tsx

```diff
@@ -38,10 +33,18 @@ const propTypes = {
   children: PropTypes.node.isRequired,
   className: PropTypes.string,
   title: PropTypes.string.isRequired,
-  type: PropTypes.oneOf<AlertBoxType>(["warning", "success", "error", "info"]),
+  type: PropTypes.oneOf<AlertBoxType>(["processing", "warning", "success", "error", "info"]),
   isClosable: PropTypes.bool,
 };
 
+const iconMap = {
+  processing: "hourglass-half",
+  warning: "exclamation-triangle",
+  success: "thumbs-up",
+  error: "exclamation-circle",
+  info: "bell",
+} as const;
```

Comment:
I'm not familiar with the `as const` declaration. What does it do here?

Could we declare our own `IconMap` type instead?

```ts
type IconMap = {
  [x in AlertBoxType]: string;
};

const iconMap: IconMap = {
  processing: "hourglass-half",
  warning: "exclamation-triangle",
  success: "thumbs-up",
  error: "exclamation-circle",
  info: "bell"
};
```

Reply:
`as const` is a relatively new TypeScript feature called const assertion. It makes the types narrower, which in this case restricts the values in the object to be in a set of string literals instead of `string`. `FontAwesome` accepts `string` as its `name` prop, so this isn't providing any real benefit here. If the `name` prop only accepted certain string literals, then the const assertion would prevent a compiler error. 

I'll just remove it for simplicity since it's not needed.